crypto/x509/by_dir.c

```diff
@@ -257,165 +258,180 @@ static int get_cert_by_subject_ex(X509_LOOKUP *xl, X509_LOOKUP_TYPE type,
+                    goto finish;
+                idx = sk_BY_DIR_HASH_find(ent->hashes, &htmp);
+                if (idx >= 0) {
+                    hent = sk_BY_DIR_HASH_value(ent->hashes, idx);
+                    k = hent->suffix;
+                } else {
+                    hent = NULL;
+                    k = 0;
+                }
+                CRYPTO_THREAD_unlock(ctx->lock);
             } else {
-                c = '\0';
+                k = 0;
+                hent = NULL;
             }
+            for (;;) {
+                char c = '/';
+
+    #ifdef OPENSSL_SYS_VMS
+                c = ent->dir[strlen(ent->dir) - 1];
+                if (c != ':' && c != '>' && c != ']') {
+                    /*
+                    * If no separator is present, we assume the directory
+                    * specifier is a logical name, and add a colon.  We really
+                    * should use better VMS routines for merging things like
+                    * this, but this will do for now... -- Richard Levitte
+                    */
+                    c = ':';
+                } else {
+                    c = '\0';
+                }
 
-            if (c == '\0') {
-                /*
-                 * This is special.  When c == '\0', no directory separator
-                 * should be added.
-                 */
-                BIO_snprintf(b->data, b->max,
-                             "%s%08lx.%s%d", ent->dir, h, postfix, k);
-            } else
-#endif
-            {
-                BIO_snprintf(b->data, b->max,
-                             "%s%c%08lx.%s%d", ent->dir, c, h, postfix, k);
-            }
-#ifndef OPENSSL_NO_POSIX_IO
-# ifdef _WIN32
-#  define stat _stat
-# endif
-            {
-                struct stat st;
-                if (stat(b->data, &st) < 0)
-                    break;
-            }
-#endif
-            /* found one. */
-            if (type == X509_LU_X509) {
-                if ((X509_load_cert_file_ex(xl, b->data, ent->dir_type, libctx,
-                                            propq)) == 0)
-                    break;
-            } else if (type == X509_LU_CRL) {
-                if ((X509_load_crl_file(xl, b->data, ent->dir_type)) == 0)
-                    break;
+                if (c == '\0') {
+                    /*
+                    * This is special.  When c == '\0', no directory separator
+                    * should be added.
+                    */
+                    BIO_snprintf(b->data, b->max,
+                                "%s%08lx.%s%d", ent->dir, h, postfix, k);
+                } else
+    #endif
+                {
+                    BIO_snprintf(b->data, b->max,
+                                "%s%c%08lx.%s%d", ent->dir, c, h, postfix, k);
+                }
+    #ifndef OPENSSL_NO_POSIX_IO
+    # ifdef _WIN32
+    #  define stat _stat
+    # endif
+                {
+                    struct stat st;
+                    if (stat(b->data, &st) < 0)
```

Comment:
As you're touching this (although it was preexisting) - please add one empty line after declarations.

Reply:
ef74616d2ea2cafcfe6c9cb04e00b69b0f6e8c6c